app/jobs/catalogue_check_job.rb

```diff
@@ -0,0 +1,76 @@
+class CatalogueCheckWorker < ActiveJob::Base
+  queue_as :default
+
+  require 'capybara'
+  require 'capybara/poltergeist'
+  require_relative '../../app/jobs/scrape_job.rb'
+  require_relative '../../app/jobs/delete_job.rb'
+
+  include Capybara::DSL
+  Capybara.register_driver :poltergeist do |app|
+    Capybara::Poltergeist::Driver.new(app, timeout: 60, js_errors: false)
+  end
+  Capybara.default_driver = :poltergeist
+
+  def perform
+
+    #setting the initial variable
+    current_catalogue = Hash.new(0)
+    compare_catalogue_array = []
+    new_catalogue = []
+    catalogue_delete = []
+    region_array = ["Melbourne, 3000", "Sydney, 2000"]
+
+    #collect current_catalogue data

```

Comment:
The readability of the catalogue processing code here could be improved. It's a little hard to follow exactly what's happening. Once there are some tests in place for this code, it'd be a good candidate for refactoring. 

I like what you've done with extracting the `catalogue_gather` method. You can extract more methods here and give them good names, that'd help readability and avoid the need for the comments. Aim to have well named methods and variables to avoid the need for explanatory comments.